src/wallet/rpcwallet.cpp

```diff
@@ -4216,6 +4216,55 @@ UniValue getwalletinfo(const JSONRPCRequest& request)
     return obj;
 }
 
+UniValue getcommunityinfo(const JSONRPCRequest& request)
+{
+    CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() > 1)
+        throw std::runtime_error(
+            "getwalletinfo ( \"address\" )\n"
+            "Returns an object containing various community state info.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"referralcount\": xxxx,             (numeric) number of confirmed referrals\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getcommunityinfo", "")
+            + HelpExampleRpc("getcommunityinfo", "")
+        );
+
+    ObserveSafeMode();
+    LOCK2(cs_main, pwallet->cs_wallet);
+
+    UniValue obj(UniValue::VOBJ);
+
+    uint160 address;
+    if (!request.params[0].isNull()) {
+        CMeritAddress dest(request.params[0].get_str());
```

Comment:
Use the function LookupDestination here, which will support address lookup by alias.